### PR TITLE
Integrate Wolframscript Execution into WallGo and Update Flag Usage

### PR DESCRIPTION
**Pull request by pschicho**
This PR introduces the ability to run Wolframscript directly from within WallGo.

Key changes:
- Command-Line Script: Added a script that allows the path to be passed directly, facilitating the execution of Wolframscript. This requires the changes introduced in [WallGoMatrix PR28](https://github.com/Wall-Go/WallGoMatrix/pull/28).
- Flag Update: Modified the usage of the --recalculateMatrixElements flag, enabling the execution of a specified .m file and allowing users to define the output file(s).

Testing:
- Needs testing on multiple platforms to ensure cross-platform compatibility.

### CODE REVIEW

**Comment by jorindevandevis** [comment]
I tried it, and I got this error:

Fatal: Error when generating matrix elements from mathematica via WallGoMatrix

Your Wolfram product is not activated or is experiencing a license-related problem.

Please activate the product at the following WolframKernel location:

    /Applications/Mathematica.app/Contents/MacOS/WolframKernel

**Comment by jorindevandevis** [comment]
Got the WolframKernel to work eventually (which was a bit non-trivial). 
I suggest to put in the readme of WallGoMatrix that WolframKernel needs to be activated for this functionality, as well as in the code (where the command line argument is defined).
Also WallGoMatrix now prints an enormous amount of numbers to my terminal (we discussed this earlier). Is it possible to suppress that?

**Comment by jorindevandevis** [comment]
Great, output looks very good now! Maybe remove "using WallGoMatrix" from

    Matrix elements are being stored at 
    Output path: /Users/..../MatrixElements/UserGenerated
    using WallGoMatrix.

File: src/WallGo/mathematicaHelpers.py
Comment (niemilau) [inline src/WallGo/mathematicaHelpers.py:20]: What does "active WolframKernel" mean? I'm under the impression that wolframscript starts a new kernel when invoked.

**Comment by niemilau** [comment]
What is the motivation for having matrix element generation built in to the base example? To me it seems unnecessary since we provide matrix elements anyway, and producing different ones dynamically at runtime would require runtime changes to the .m model files themselves.

The generateMatrixElementsViaSubprocess() function itself seems to work on both ubuntu and Windows. But there are some issues related to installation and running the .m in script mode which I wrote in https://github.com/Wall-Go/WallGoMatrix/pull/28#issuecomment-2419820026.

File: src/WallGo/mathematicaHelpers.py
Comment (jorindevandevis) [reply src/WallGo/mathematicaHelpers.py:20]: I think it's meant that it has to be "activated" (I got an error that mine wasn't)

File: src/WallGo/mathematicaHelpers.py
Comment (niemilau) [inline src/WallGo/mathematicaHelpers.py:49]: Not a fan of having a long print like this. How about just one line: "Generating matrix elements from {inFilePath} to {outFilePath}" ?

File: src/WallGo/mathematicaHelpers.py
Comment (pschicho) [reply src/WallGo/mathematicaHelpers.py:20]: Isn't that the same thing?

File: src/WallGo/mathematicaHelpers.py
Comment (pschicho) [reply src/WallGo/mathematicaHelpers.py:49]: Needed to make this distinct from the WallGo python output. But I can make the printout smaller.

File: src/WallGo/mathematicaHelpers.py
Comment (pschicho) [reply src/WallGo/mathematicaHelpers.py:66]: The commented line only surpasses the long print outs of WallGoMatrix. The error is only a subprocess one. And probably errors within the wolframscript run are not triggered indeed. I will try to cut down the output of this from within WallGoMatrix.

File: src/WallGo/mathematicaHelpers.py
Comment (niemilau) [inline src/WallGo/mathematicaHelpers.py:56]: These seem unnecessary for now at least since its the same for all platforms

File: src/WallGo/mathematicaHelpers.py
Comment (niemilau) [reply src/WallGo/mathematicaHelpers.py:66]: I propose having a descriptive error here. Something saying that the user should ensure they have WolframEngine installed and licensed.

File: src/WallGo/mathematicaHelpers.py
Comment (niemilau) [reply src/WallGo/mathematicaHelpers.py:20]: Active kernel sounds like it needs to be running prior to calling this function. IMO the description should say that you need to have a licensed installation of WolframEngine.

File: Models/wallGoExampleBase.py
Comment (niemilau) [inline Models/wallGoExampleBase.py:232]: Move import to file top

File: Models/wallGoExampleBase.py
Comment (niemilau) [inline Models/wallGoExampleBase.py:237]: licensed WolframEngine

**Review by niemilau — COMMENTED** [review]
Let's wait until the PR on WallGoMatrix is merged

**Review by niemilau — APPROVED** [review]

**Review by niemilau — APPROVED** [review]

**Pull request merged**
